pkg/registries/filter.go

```diff
@@ -0,0 +1,227 @@
+package registries
+
+import (
+	"regexp"
+	"sync"
+)
+
+type filterMode uint8
+
+const (
+	filterModeBoth filterMode = iota
+	filterModeWhite
+	filterModeBlack
+	filterModeNone
+)
+
+type failedRegexp struct {
```

Comment:
Kubernetes seems to use the type file. the standard lib does not. a good example is the net/http package. https://golang.org/src/net/http/header.go. Usually, when talking about good style/idioms  I side with the standard library.